packages/connection-model/lib/model.js

```diff
@@ -805,11 +817,42 @@ Connection = AmpersandModel.extend({
     }
 
     if (attrs.authStrategy === 'KERBEROS') {
-      if (!attrs.kerberosServiceName) {
-        attrs.kerberosServiceName = KERBEROS_SERVICE_NAME_DEFAULT;
-      }
+      const authProperties = attrs.authMechanismProperties || {};
+      this.kerberosServiceName = defaultTo(
+        attrs.kerberosServiceName,
+        defaultTo(attrs.gssapiServiceName, authProperties.SERVICE_NAME)
+      );
+
+      this.kerberosServiceRealm = defaultTo(
+        attrs.kerberosServiceRealm,
+        defaultTo(attrs.gssapiServiceRealm, authProperties.SERVICE_REALM)
+      );
 
-      this.kerberosServiceName = attrs.kerberosServiceName;
+      this.kerberosCanonicalizeHostname = defaultTo(
```

Comment:
Is it possible to get all of those variants here? Isn't attrs just what we get from the form?

Reply:
That's basically everything I found being used. So:
* `attrs.kerberosCanonicalizeHostname` - that's essentially what we store in the model ourselves
* `attrs.gssapiCanonicalizeHostName` - that's a property that has existed next to `gssapiServiceName` on the model for a long time, so I'm just taking care of that. Seems to have been be mapped by `gssapiCanonicalizeHostName` query parameter (not documented)
* `attrs.gssapicanonicalizehostname` - this is the property you _actually_ get when using `?gssapiCanonicalizeHostName=true` for _whatever_ reason from the parseConnectionString method
* `authProperties.CANONICALIZE_HOST_NAME` - the "official" property according to the spec
* `authProperties.gssapiCanonicalizeHostName` - the "unofficial" property that the driver _actually_ looks at.